server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -71,7 +72,7 @@ protected void isValidUsername(String username) throws UsersRepositoryException
         int i = username.indexOf("@");
         if (supportVirtualHosting()) {
             // need a @ in the username
-            if (!user.hasDomainPart()) {
+            if (i == -1) {
```

Comment:
why is this back?

Reply:
Because user has no DomainPart on the latest version of that PR.

I will re-add something similar.